app/src/main/java/com/example/myapplication/NewOrderDialog.java

```diff
@@ -53,7 +53,13 @@ public void onTimeChanged(TimePicker timePicker, int hour, int minute) {
                 updateOrderInstructions(hour, minute);
             }
         });
+        setButtonListeners();
+        okButton.setClickable(false);
 
+        return builder.create();
+    }
+
+    private void setButtonListeners() {
```

Comment:
I wouldn't think it is nessacery to create a function especially for that since it might be more complicated to understand, But It's just my personal flavor for how it should look. good job.

Reply:
Interesting. I'll take that under advisement